compiler/vm/src/heap/object.rs

```diff
@@ -299,7 +316,9 @@ impl Pointer {
         match &heap.get(*self).data {
             Data::Int(int) => format!("{}", int.value),
             Data::Text(text) => format!("\"{}\"", text.value),
-            Data::Symbol(symbol) => symbol.value.to_string(),
+            Data::Tag(Tag { symbol, value }) => value.map_or(symbol.to_string(), |value| {
+                format!("{}({})", symbol, value.format_helper(heap, is_debug))
```

Comment:
To stay closer to the source code, I'd add a whitespace before the parentheses. We can add a smarter stringification later on that tracks precedence so that parentheses are only added where necessary

```suggestion
                format!("{} ({})", symbol, value.format_helper(heap, is_debug))
```